data/features/featureList.yaml

```diff
@@ -0,0 +1,891 @@
+## Annotated examples
+## Just for documentation, not used by code. Please leave intact.
+examples:
+
+  # A feature gate that is alpha, enabled, and still configurable
+- name: ExampleFeature1
+  alpha:
+    fromVersion: v1.3
+    state: true # override the default state for this stage (false)
+  deprecated:
+    fromVersion: v1.18
+
+  # An experiment that didn't work out. After a while in alpha, it
+  # was deprecated and can no longer be enabled.
+- name: ExampleFeature2
+  alpha:
+    fromVersion: v1.15
+  deprecated:
+    fromVersion: v1.18
+
+  # A feature gate that is no longer available. This started as a
+  # beta featured but, unusually, one that was initially disabled
+  # by default.
+  # Later, it was enabled by default (still beta), eventually
+  # graduating to general availability.
+- name: ExampleFeature3
+  beta:
```

Comment:
Hi @sftim . I am looking over the data format and the parsing. Some feedback:

I noticed that there seems to be a greater interest in maintaining these tables, typically at release time.
Previously, the docs reviewers were a bit more relaxed about enforcing the updates (the information eventually was updated). Now, updating the feature gates seems to be more of a priority.

Anyway, my biggest concern with these changes is **usability**.
Is the data structure easy to understand? If the data structure is not correct, will the program/publisher fail?

I agree that generating these tables from a data file makes sense if the data is derived directly (generated) from the source code. I don't know if this would be possible.

I think managing and asking contributors to edit the YAML is probably more difficult than editing a Markdown table.
The Markdown table has visual hints that make it somewhat easier to document the data.
That said, I realize that even the Markdown tables can be confusing (for example, what are the columns 'Since' and 'Until'). And the contributor has to fiddle with the Markdown (formatting the text).
However, if a mistake is made in the table structure, the table fails to render. We don't check the data values.

If I were designing  the feature gate data, I would make it very explicit, in other words,
- don't prefix every version number with 'v'
- define the 'From' and 'To' or 'Since' and 'Until' version numbers. Don't leave these decisions to the parser/program processing the data. What you define in the data file is what appears in the table.
- Would it help users edit the data if the data was split over two files?

So, is the data file easier to understand and use over the Markdown table?
Does it help maintain the content?

Reply:
> I think managing and asking contributors to edit the YAML is probably more difficult than editing a Markdown table.

Honestly, people who work on Kubernetes code are likely to know YAML. Maybe I'm over-estimating but YAML is a big part of how folks I know interact with Kubernetes.

> don't prefix every version number with 'v'
YAML will then treat these strings as numbers. The `v` prefix means we don't need to quote them.

> We don't check the data values.

For me, that's a benefit of the switch. If you propose an invalid change, we can fail that during CI. This YAML format makes it _much_ easier for a machine / computer program when you made a mistake. I want those failures to occur as early as possible, in other words to spot them when the PR is opened rather than later, perhaps even post release.

Reply:
I tried a new approach in https://github.com/kubernetes/website/pull/41793